cmd/kubelet/app/server.go

```diff
@@ -124,27 +120,75 @@ is checked every 20 seconds (also configurable with a flag).
 
 HTTP server: The kubelet can also listen for HTTP and respond to a simple API
 (underspec'd currently) to submit a new manifest.`,
+		// The Kubelet has special flag parsing requirements to enforce flag precedence rules,
+		// so we do all our parsing manually in Run, below.
+		// DisableFlagParsing=true provides the full set of flags passed to the kubelet in the
+		// `args` arg to Run, without Cobra's interference.
+		DisableFlagParsing: true,
```

Comment:
does it strip any parent command args (e.g. `hyperkube kubelet`) or does it leave us with essentially `os.Args[1:]`?

Reply:
I think it strips parent command args, based on my reading of the cobra internals, but I still have to write a small test program to confirm.

Reply:
It strips parent command names.

Test program:
```
package main

import (
	"fmt"
	"os"

	"github.com/spf13/cobra"
)

func main() {
	parent := &cobra.Command{
		Use: "parent",
		Run: func(cmd *cobra.Command, args []string) {},
	}
	child := &cobra.Command{
		Use:                "child",
		DisableFlagParsing: true,
		Run: func(cmd *cobra.Command, args []string) {
			fmt.Println(args)
		},
	}
	parent.AddCommand(child)
	if err := parent.Execute(); err != nil {
		fmt.Fprintf(os.Stderr, "%v\n", err)
		os.Exit(1)
	}
}
```

Output:
```
$ ./parent child --foo
[--foo]
```